bite-size-notes/week-4-e-stochastic bellman equation.md

```diff
@@ -0,0 +1,167 @@
+
+---
+## Stochastic Bellman value function Formation
+As following the same process of the deterministic value function we learned from the lecture, given an initial state $x_0$ and a policy $\pi = {\mu_0,..\mu_{N-1}}$, the future states $x_k$ and disturbances $\omega_k$ are random variables with probability distributions(i.e Guassian Normal Distribution) defined through the system equation,
+
+$$
+     x_{k+1} = f_k(x_k, \mu_k(x_k), \omega_k), \quad k=0,1,...,N-1
+$$
+
+Policies are more general objects than control sequences, and in the presence of stochastic uncertainty, they can result in improved cost, since they allow choices of controls $u_k$ that incorporate knowledge of the state $x_k$.
+
+Thus, for given functions $g_k$, $k=0,1,..., N$, the expected cost of the policy $\pi$ starting $x_k$ is :
+
+$$
+   V_{\pi}(x_k) = \mathbb{E} \[ g_N(x_N) + \sum^{N-1}_{m= k} g_k(x_m, \mu_k(x_m), \omega_m) \]                
+$$
+
+where the expected value operation  $\mathbb{E}\{\cdot\}$ is over all the random variables $\omega_k$ and $x_k$. An optimal policy $\pi^*$ is one that minimizes this cost; i.e,
+
+$$
+    V_{\pi^*}(x_k) = \min_{u_m \in U_m(x_m), m=k,..,N-1} \mathbb{E} \[ g_N(x_N) + \sum_{m=k}^{N-1} g_m(x_m, \mu_k(x_m), \omega_m) \] 
+$$
+
+$$
+    = \min_{u_k \in U_k(x_k)} \mathbb{E} \[ g_k(x_k) + \min_{u_m \in U_m(x_m), m=k+1,..,N-1} \[g_m(x_m) + \sum_{m=k+1}^{N-1} g_m(x_m, \mu_k(x_m), \omega_m) \] \]
+$$
+
+$$
+    = \min_{u_k \in U_k(x_k)} \mathbb{E} \[ g_k(x_k) + V_{\pi^*}(x_{k+1}) \]
+$$
+
+$$
+    = \min_{u_k \in U_k(x_k)} \mathbb{E} \[ g_k(x_k) + V_{\pi^*}(f_k(x_k, \mu_k(x_k), \omega_k)) \]
+$$
+
+$$
+    = \min_{\pi\in\Pi}V_\pi(x_k)
+$$
+
+where $V_\pi(x_k) = \mathbb{E} \[ g_k(x_k) + V_{\pi^*}(x_{k+1}) \]$ referring __one-step look ahead optimal cost__ and $\Pi$ is the set of all __admissible__ policies at state $x_k$.
+
+The optimal cost depends on $x_k$ and is denoted by $V^*(x_k)$; 
+
+$$
+    V^*(x_k) = \min_{\pi \in \Pi} V_{\pi}(x_k)
+$$
+
+It is useful to view $V^\*$ as a function that assigns to each current state $x_k$ the optimal cost $V^*(x_k)$, and call it the __optimal Bellman cost function or optimal Bellman value function__, particularly in problems of maximizing the reward.
+
+---
+## Finite Horizon Stochastic Dynamic Programming
+
+Optimization problems such as the one stated above are efficiently solved via dynamic programming (DP). DP relies on the following obvious fact: if a given state-control action sequence is optimal, and we were to remove the first state and control action, the remaining sequence is also optimal (with the second state of the original sequence now acting as initial state). This is the Bellman optimality principle. The optimality principle can be revised in similar language: the choice of optimal actions in the future is independent of the past actions which led to the present state. This property resembles the markov assumption.

```

Comment:
I like you introduce this idea to explain why PD is more efficient than direct method.